subprojects/packagefiles/libgpiod/lib/meson.build

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# Copyright 2021 IBM Corp.
+
+gpiod_sources = [ 'core.c', 'ctxless.c', 'helpers.c', 'iter.c', 'misc.c' ]
+gpiod = library('gpiod', gpiod_sources,
+                include_directories: gpiod_includes,
+                gnu_symbol_visibility: 'hidden',
+                c_args: [ '-include', '../config.h' ],
+                version: abi_gpiod_version,
+                install: true)
+
+pkg.generate(gpiod,
+             name: 'libgpiod',
+             description: 'Library and tools for the Linux GPIO chardev')
+
+gpiod_dep = declare_dependency(link_with: gpiod,
+                               include_directories: gpiod_includes,
+                               version: abi_gpiod_version)
+meson.override_dependency('libgpiod', gpiod_dep)
```

Comment:
This requires meson 0.54, preventing use with older meson versions, e.g. it will not work with the meson provided by debian stable or Ubuntu 20.04 LTS.

Instead, you can use
```ini
[provide]
libgpiod = gpiod_dep
```

Which will do the same thing as dependency_names but without meson.override_dependency

Later calls to meson.override_find_program don't have a backwards-compatible declaration form, but they are also probably not needed. Are any of them actually going to be used in other projects depending on libgpiod via `find_program('gpioinfo')` or whatnot?

By dropping the use of meson.override_*, you can get down to a minimum meson version of 0.48.0, which can be declared in `project(..., meson_version: '>=0.48.0')`

Reply:
Thanks for relaxing the 90.54 requirement. Personally, I'd still like to explicitly list the minimum requirement though:

```meson
project('libgpiod', 'c', 'cpp',
        ...,
        meson_version: '>=0.48.0')
```

It serves as useful documentation of expectations, and raises warnings if someone tries to update it to require functions from newer versions, so that doesn't happen by accident.